clippy_lints/src/methods/mod.rs

```diff
@@ -1383,6 +1383,38 @@ declare_clippy_lint! {
     "using unnecessary lazy evaluation, which can be replaced with simpler eager evaluation"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for `from_iter()` function calls that implements `FromIterator`
+    /// trait.
+    ///
+    /// **Why is this bad?** Makes code less readable especially in method chaining.
```

Comment:
I don't think it makes code less readable. It is just the recommended style to use `collect`, like it is documented in the `FromIterator` documentation: https://doc.rust-lang.org/std/iter/trait.FromIterator.html

Please also add a reference to this documentation to "Why is this bad?"